src/coreclr/jit/lclvars.cpp

```diff
@@ -3493,13 +3493,42 @@ void Compiler::lvaSortByRefCount()
             {
                 varDsc->lvTracked = 0;
             }
-            else if ((varDsc->lvType == TYP_STRUCT) && !varDsc->lvRegStruct && !compEnregStructLocals())
+            else if (!varDsc->IsEnregisterableType())
             {
                 lvaSetVarDoNotEnregister(lclNum DEBUGARG(DNER_IsStruct));
             }
-            else if (!varDsc->IsEnregisterableType())
+            else if (varDsc->lvType == TYP_STRUCT)
             {
-                lvaSetVarDoNotEnregister(lclNum DEBUGARG(DNER_IsStruct));
+                if (!varDsc->lvRegStruct && !compEnregStructLocals())
+                {
+                    lvaSetVarDoNotEnregister(lclNum DEBUGARG(DNER_IsStruct));
+                }
+                else if (varDsc->lvIsMultiRegArgOrRet())
+                {
+                    // Prolog and return generators do not support SIMD<->general register moves.
+                    lvaSetVarDoNotEnregister(lclNum DEBUGARG(DNER_IsStructArg));
+                }
+            }
+            else if (varDsc->lvType == TYP_STRUCT)
```

Comment:
This is the same condition as before; this block appears to be dead code.

Reply:
looks like a merge issue and that is why I am seeing some failures in CI, will fix it soon.

Reply:
fixed, thank you.